components/brave_rewards/browser/rewards_service_impl.cc

```diff
@@ -1651,7 +1650,7 @@ bool RewardsServiceImpl::ShouldShowOnboarding() const {
     ads_supported = ads_service->IsSupportedLocale();
   }
 
-  return onboard_time.is_null() && !ads_enabled && ads_supported;
+  return !legacy_enabled && !ads_enabled && ads_supported;
```

Comment:
does this mean that a user who has previously enabled awards but not ads will get AC automatically enabled when they click through onboarding?

Reply:
nvm misread this!

Reply:
@diracdeltas what you describe is the current behavior that this PR attempts to fix.